spec/services/learning_units/resource_sorter_service_spec.rb

```diff
@@ -0,0 +1,88 @@
+require 'rails_helper'
+require 'bigdecimal/util'
+
+RSpec.describe LearningUnits::ResourceSorterService do
+  let(:first_user) { create(:user) }
+  let(:second_user) { create(:user) }
+  let(:third_user) { create(:user) }
+  let(:learning_unit) { create(:learning_unit) }
+  let(:service) { described_class.new(learning_unit) }
+  let!(:first_resource) { create(:resource, learning_unit:, user: first_user) }
+  let!(:second_resource) do
+    create(:resource, learning_unit:, user: second_user)
+  end
+  let!(:third_resource) { create(:resource, learning_unit:, user: first_user) }
+
+  describe '#sorted_resources' do
+    def create_evaluation(user, resource, evaluation)
+      create(
+        :resource_evaluation,
+        user:,
+        resource:,
+        evaluation:
+      )
+    end
+
+    context 'when there are many evaluations for many resources' do
+      before do
```

Comment:
I would transfer the creation of many evaluation to a special method on the resource factory. Something like create_resource_with_may_evaluations. 
However, considering that we have 2 hours left. I approve this test.